sdcm/scan_operation_thread.py

```diff
@@ -490,17 +489,30 @@ def _validate_fullscan_result(
         output = "\n".join([str(i) for i in result])
         if int(result[0].count) <= 0:
             return f"Fullscan failed - count is not bigger than 0: {output}", Severity.ERROR
-        get_query_trace = cmd_result.get_query_trace().events
-        for trace_event in get_query_trace:
-            if dispatch_forward_statement_regex_pattern.search(str(trace_event)):
-                regex_found = True
-                break
         self.log.debug("Fullscan aggregation result: %s", output)
 
-        if not regex_found:
-            self.log.debug("\n".join([str(i) for i in get_query_trace]))
+        prometheus = PrometheusDBStats(
+            self.fullscan_params.db_cluster.test_config.tester_obj().monitors.nodes[0].external_address)
+        node_procs_blocked = 'scylla_forward_service_requests_dispatched_to_other_nodes'
+        node_procs_res = prometheus.query(node_procs_blocked, start_execution_time, end_execution_time)
+
+        is_dispatched_forward_request = False
+        # expected format is :
+        # [{'metric': {'__name__': 'scylla_forward_service_requests_dispatched_to_other_nodes',
+        # 'instance': '10.4.0.181', 'job': 'scylla', 'shard': '0'}, 'values': [[1690287334.763, '0']]}, ...]
+        for res in node_procs_res:
+            for value in res['values']:
+                if int(value[1]) > 0:
+                    self.log.info('forward_service_requests_dispatched_to_other_nodes was triggered')
+                    is_dispatched_forward_request = True
+                    break
+            if is_dispatched_forward_request:
+                break
+
+        if not is_dispatched_forward_request:
+            self.log.debug("\n".join([str(i) for i in node_procs_res]))
             self.log.debug("Fullscan aggregation result: %s", output)
-            return "Fullscan failed - 'Dispatching forward_request' message was not found in query trace events", \
+            return "Fullscan failed - 'forward_service_requests_dispatched_to_other_nodes' was not found in query trace events", \
                 Severity.WARNING
```

Comment:
why not error ?

Reply:
good question. it was initially a WARNING. if you want set ERROR I agree with you and will set + test it

Reply:
tested. there is no 'forward_service_requests_dispatched_to_other_nodes'  at the beginning of the test. will continue investigate tomorrow

Reply:
@fruch,
Result:
1) forward_service_requests_dispatched_to_other_nodes - counter metric ( increase value when request forwarded) 
2) it updated not periodically(not immediately) with minimal increment 5 (according to what I see)

solution:
verify that matric changed every 5th aggregation scan